### PR TITLE
 Use separated methods in Forum Notifications Cron Job 

### PR DESCRIPTION
**Pull request by legionth**
Use separated methods to create the different notification types.

Also eliminates some code duplications.

### CODE REVIEW

**Comment by legionth** [comment]
Have a look @mjansenDatabay

File: Modules/Forum/classes/class.ilForumCronNotification.php [hunk 1]
```diff
@@ -48,13 +48,21 @@ class ilForumCronNotification extends ilCronJob
 	 * @var int
 	 */
 	protected $num_sent_messages = 0;
-	
+
+	/** @var ilDb */
```
Comment (mjansenDatabay) [inline Modules/Forum/classes/class.ilForumCronNotification.php]: `\ilDBInterface`

File: Modules/Forum/classes/class.ilForumCronNotification.php [hunk 2]
```diff
@@ -48,13 +48,21 @@ class ilForumCronNotification extends ilCronJob
...
+	private $ilDB;
 	/**
 	 *
 	 */
-	public function __construct()
+	public function __construct(\ilDB $database = null)
```
Comment (mjansenDatabay) [inline Modules/Forum/classes/class.ilForumCronNotification.php]: `\ilDBInterface`

**Review by mjansenDatabay — CHANGES_REQUESTED** [review]

**Comment by legionth** [comment]
Fixed the code by your remarks @mjansenDatabay

**Review by mjansenDatabay — APPROVED** [review]

**Pull request merged**